ddpui/auth.py

```diff
@@ -38,6 +46,32 @@ def authenticate(self, request, token):
         raise HttpError(400, UNAUTHORIZED)
 
 
+def has_permission(permission_slugs: list):
+    def decorator(api_endpoint):
+        @wraps(api_endpoint)
+        def wrapper(request, **kwargs):
+            # request will have set of permissions that are allowed
+            # check if permission_slug lies in this set
+            # throw error if nots
+            try:
+                if (
+                    request.permissions
+                    and len(
+                        set(permission_slugs).intersection(set(request.permissions))
```

Comment:
the request.permissions should be a subset of the permissions passed to the decorator

Reply:
Agreed. made the changes